src/main/java/com/configcat/ConfigCatClient.java

```diff
@@ -502,26 +502,26 @@ private <T> T getValueFromSettingsMap(Class<T> classOfT, SettingResult settingRe
         try {
             Map<String, Setting> settings = settingResult.settings();
             if (settings.isEmpty()) {
-                String error = "Config JSON is not present. Returning defaultValue: [" + defaultValue + "].";
+                String error = "Config JSON is not present. Returning the `defaultValue` parameter that you specified in your application: '" + defaultValue + "'.";
                 this.hooks.invokeOnFlagEvaluated(EvaluationDetails.fromError(key, defaultValue, error, userObject));
-                this.logger.error(error);
+                this.logger.error(1000, errorMessage);
                 return defaultValue;
             }
 
             Setting setting = settings.get(key);
             if (setting == null) {
-                String error = "Value not found for key " + key + ". Here are the available keys: " + String.join(", ", settings.keySet());
+                String error = "Failed to evaluate setting '" + key + "' (the key was not found in config JSON). Returning the `defaultValue` parameter that you specified in your application: '" + defaultValue + "'. Available keys: " + String.join(", ", settings.keySet()) + ".";
```

Comment:
Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`